### PR TITLE
feat: Support micro-frontends in app layout

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Remove dependency on `AppLayoutContext`, introduce CSS-vars without hashes. These become our de-facto public API

Related links, issue #, if available: n/a

### How has this been tested?

Screenshot tests in my dev pipeline

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/styles/global-vars.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/*
+ Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+$stickyVerticalTopOffset: --awsui-sticky-vertical-top-offset;
+$stickyVerticalBottomOffset: --awsui-sticky-vertical-bottom-offset;
```
Comment (just-boris) [inline src/internal/styles/global-vars.scss:7]: No code generation for this file, because introducing any global var is a big deal and it should not be easy.

Manual steps here is a feature, not an issue

File: src/internal/context/app-layout-context.ts [hunk 2]
```diff
@@ -3,17 +3,10 @@
 import { useContext, createContext } from 'react';
 
 export interface AppLayoutContextProps {
-  stickyOffsetBottom: number;
-  stickyOffsetTop: number;
-  mobileBarHeight?: number;
   setHasStickyBackground?: (hasBackground: boolean) => void;
```
Comment (just-boris) [inline src/internal/context/app-layout-context.ts:6]: Cannot remove this context entirely, because of this dependency. 

It will be gone in a separate PR with the entire dark header feature

File: src/header/internal.tsx [hunk 3]
```diff
@@ -55,6 +57,7 @@ export default function InternalHeader({
         baseProps.className,
         styles[`root-variant-${variantOverride}`],
         isRefresh && styles.refresh,
+        hasToolbarHeader && styles['root-with-toolbar'],
```
Comment (just-boris) [inline src/header/internal.tsx:60]: First design deviation for the toolbar feature flag. More to come in the future PRs

File: src/app-layout/classic.tsx [hunk 4]
```diff
@@ -500,20 +499,16 @@ const ClassicAppLayout = React.forwardRef(
...
+                  `${stickyNotificationsHeight}px`,
+                  mobileBarHeight && !disableBodyScroll ? `${mobileBarHeight}px` : '0px',
+                  !!disableBodyScroll,
+                  isMobile
+                )}
               >
-                <AppLayoutContext.Provider
-                  value={{
-                    stickyOffsetTop:
-                      // We don't support the table header being sticky in case the deprecated disableBodyScroll is enabled,
```
Comment (just-boris) [inline src/app-layout/classic.tsx:507]: Fun fact. Now we do. It was easier to support this feature than to port hacks into the new code

File: src/internal/generated/styles/tokens.d.ts [hunk 5]
```diff
@@ -61,3 +61,9 @@ export const colorChartsThresholdNegative: string;
 export const colorChartsThresholdPositive: string;
 export const colorChartsThresholdInfo: string;
 export const colorChartsThresholdNeutral: string;
+
+// Spacing
+export const spaceXs: string;
```
Comment (just-boris) [inline src/internal/generated/styles/tokens.d.ts:66]: We maintain a manually crafted `*.d.ts` declaration for our tokens, because the real token generation runs later in our build tasks. To break this circle, this file exists to make typescript build pass without dependency on style-dictionary -> theming build

**Review by michaeldowseza — APPROVED** [review]
LGTM!
